### PR TITLE
Allow emoji short code in labels

### PR DESCRIPTION
**Pull request by mrsdizzie**
As title, turn :alias: type short code into emojis when rendering labels to match previous behavior since this was considered regression

### CODE REVIEW

**Review by jolheiser — COMMENTED** [review]
I'm not sure it would be ideal to _replace_ the `Name` field with the rendered content. It seems like that could cause some confusing states, especially since it isn't idempotent.

Maybe instead a helper func `label.RenderName()` that could be used in templates or wherever it needs to be rendered?

**Comment by mrsdizzie** [comment]
@jolheiser I did think about doing it that way but this loaded content is only used to pass to templates and it felt a bit better and cleaner than adding more text to those. 

For honesty I don't personally like this feature much and think it is pretty niche to want to use a short code thats several times longer than just using the emoji itself (which is always supported) in something very text limited such as a label, so it felt a bit bad to have to add even slightly more complication to many templates just to support that -- some of these lines are almost 500 columns wide already 😬

But that aside there is already a helper function that would do this in a template so changing it in template could be done very easily if that is more preferred : )

**Comment by lafriks** [comment]
I don't also like that model value is changed for display. Also models should not be depend on modules/emoji imho. I would prefer that rendering should occur in views

**Comment by mrsdizzie** [comment]
changed to do inside of templates

**Comment by silverwind** [comment]
There's quite a few more matches when I search for the `has-emoji` class in the templates. It was previously used to do emoji replacement, e.g. previous JS did: 

````js
          const hasEmoji = react.find('.has-emoji');
          for (let i = 0; i < hasEmoji.length; i++) {
            emojify.run(hasEmoji.get(i));
          }
````

I think we can remove all those classes now and do replacements inside those elements.

**Comment by mrsdizzie** [comment]
Im at my limit for anything related to this, so those could be another PR or if anybody wants to take changes from this/make a new PR with another way.

**Comment by jolheiser** [comment]
I'm fine with the rest being cleaned up in another PR. Can you remove `has-emoji` class from the ones changed in this PR for now? I don't think it controls any style, it was only used for the JS afaik.

File: templates/repo/issue/labels/label_list.tmpl [hunk 1]
```diff
@@ -31,7 +31,7 @@
 			<li class="item">
 			<div class="ui grid middle aligned">
 				<div class="four wide column">
-					<div class="ui label has-emoji" style="color: {{.ForegroundColor}}; background-color: {{.Color}}">{{svg "octicon-tag" 16}} {{.Name}}</div>
+					<div class="ui label has-emoji" style="color: {{.ForegroundColor}}; background-color: {{.Color}}">{{svg "octicon-tag" 16}} {{.Name | RenderEmojiPlain}}</div>
```
Comment (mrsdizzie) [inline templates/repo/issue/labels/label_list.tmpl]:
```suggestion
					<div class="ui label" style="color: {{.ForegroundColor}}; background-color: {{.Color}}">{{svg "octicon-tag" 16}} {{.Name | RenderEmojiPlain}}</div>
```

File: templates/repo/issue/labels/label_list.tmpl [hunk 2]
```diff
@@ -74,7 +74,7 @@
 					<li class="item">
 					<div class="ui grid middle aligned">
 						<div class="three wide column">
-							<div class="ui label has-emoji" style="color: {{.ForegroundColor}}; background-color: {{.Color}}">{{svg "octicon-tag" 16}} {{.Name}}</div>
+							<div class="ui label has-emoji" style="color: {{.ForegroundColor}}; background-color: {{.Color}}">{{svg "octicon-tag" 16}} {{.Name | RenderEmojiPlain}}</div>
```
Comment (mrsdizzie) [inline templates/repo/issue/labels/label_list.tmpl]:
```suggestion
							<div class="ui label" style="color: {{.ForegroundColor}}; background-color: {{.Color}}">{{svg "octicon-tag" 16}} {{.Name | RenderEmojiPlain}}</div>
```

File: templates/repo/issue/list.tmpl [hunk 3]
```diff
@@ -51,7 +51,7 @@
...
 							<a class="item" href="{{$.Link}}?q={{$.Keyword}}&type={{$.ViewType}}&sort={{$.SortType}}&state={{$.State}}&milestone={{$.MilestoneID}}&assignee={{$.AssigneeID}}">{{.i18n.Tr "repo.issues.filter_label_no_select"}}</a>
 							{{range .Labels}}
-								<a class="item has-emoji label-filter-item" href="{{$.Link}}?q={{$.Keyword}}&type={{$.ViewType}}&sort={{$.SortType}}&state={{$.State}}&labels={{.QueryString}}&milestone={{$.MilestoneID}}&assignee={{$.AssigneeID}}" data-label-id="{{.ID}}">{{if .IsExcluded}}{{svg "octicon-circle-slash" 16}}{{else if .IsSelected}}{{svg "octicon-check" 16}}{{end}}<span class="label color" style="background-color: {{.Color}}"></span> {{.Name}}</a>
+								<a class="item has-emoji label-filter-item" href="{{$.Link}}?q={{$.Keyword}}&type={{$.ViewType}}&sort={{$.SortType}}&state={{$.State}}&labels={{.QueryString}}&milestone={{$.MilestoneID}}&assignee={{$.AssigneeID}}" data-label-id="{{.ID}}">{{if .IsExcluded}}{{svg "octicon-circle-slash" 16}}{{else if .IsSelected}}{{svg "octicon-check" 16}}{{end}}<span class="label color" style="background-color: {{.Color}}"></span> {{.Name | RenderEmojiPlain}}</a>
```
Comment (mrsdizzie) [inline templates/repo/issue/list.tmpl]:
```suggestion
								<a class="item label-filter-item" href="{{$.Link}}?q={{$.Keyword}}&type={{$.ViewType}}&sort={{$.SortType}}&state={{$.State}}&labels={{.QueryString}}&milestone={{$.MilestoneID}}&assignee={{$.AssigneeID}}" data-label-id="{{.ID}}">{{if .IsExcluded}}{{svg "octicon-circle-slash" 16}}{{else if .IsSelected}}{{svg "octicon-check" 16}}{{end}}<span class="label color" style="background-color: {{.Color}}"></span> {{.Name | RenderEmojiPlain}}</a>
```

File: templates/repo/issue/list.tmpl [hunk 4]
```diff
@@ -155,7 +155,7 @@
 						<div class="menu">
 							{{range .Labels}}
 								<div class="item issue-action has-emoji" data-action="toggle" data-element-id="{{.ID}}" data-url="{{$.RepoLink}}/issues/labels">
```
Comment (mrsdizzie) [inline templates/repo/issue/list.tmpl]:
```suggestion
								<div class="item issue-action" data-action="toggle" data-element-id="{{.ID}}" data-url="{{$.RepoLink}}/issues/labels">
```

File: templates/repo/issue/list.tmpl [hunk 5]
```diff
@@ -220,7 +220,7 @@
 					{{end}}
 
 					{{range .Labels}}
-						<a class="ui label has-emoji" href="{{$.Link}}?q={{$.Keyword}}&type={{$.ViewType}}&state={{$.State}}&labels={{.ID}}&milestone={{$.MilestoneID}}&assignee={{$.AssigneeID}}" style="color: {{.ForegroundColor}}; background-color: {{.Color}}" title="{{.Description}}">{{.Name}}</a>
+						<a class="ui label has-emoji" href="{{$.Link}}?q={{$.Keyword}}&type={{$.ViewType}}&state={{$.State}}&labels={{.ID}}&milestone={{$.MilestoneID}}&assignee={{$.AssigneeID}}" style="color: {{.ForegroundColor}}; background-color: {{.Color}}" title="{{.Description | RenderEmojiPlain}}">{{.Name | RenderEmojiPlain}}</a>
```
Comment (mrsdizzie) [inline templates/repo/issue/list.tmpl]:
```suggestion
						<a class="ui label" href="{{$.Link}}?q={{$.Keyword}}&type={{$.ViewType}}&state={{$.State}}&labels={{.ID}}&milestone={{$.MilestoneID}}&assignee={{$.AssigneeID}}" style="color: {{.ForegroundColor}}; background-color: {{.Color}}" title="{{.Description | RenderEmojiPlain}}">{{.Name | RenderEmojiPlain}}</a>
```

File: templates/repo/issue/view_content/sidebar.tmpl [hunk 6]
```diff
@@ -111,13 +111,13 @@
 				<div class="no-select item">{{.i18n.Tr "repo.issues.new.clear_labels"}}</div>
 				{{if or .Labels .OrgLabels}}
 					{{range .Labels}}
-						<a class="{{if .IsChecked}}checked{{end}} item has-emoji" href="#" data-id="{{.ID}}" data-id-selector="#label_{{.ID}}"><span class="octicon-check {{if not .IsChecked}}invisible{{end}}">{{svg "octicon-check" 16}}</span><span class="label color" style="background-color: {{.Color}}"></span> {{.Name}}
-						{{if .Description }}<br><small class="desc">{{.Description}}</small>{{end}}</a>
+						<a class="{{if .IsChecked}}checked{{end}} item has-emoji" href="#" data-id="{{.ID}}" data-id-selector="#label_{{.ID}}"><span class="octicon-check {{if not .IsChecked}}invisible{{end}}">{{svg "octicon-check" 16}}</span><span class="label color" style="background-color: {{.Color}}"></span> {{.Name | RenderEmojiPlain}}
```
Comment (mrsdizzie) [inline templates/repo/issue/view_content/sidebar.tmpl]:
```suggestion
						<a class="{{if .IsChecked}}checked{{end}} item" href="#" data-id="{{.ID}}" data-id-selector="#label_{{.ID}}"><span class="octicon-check {{if not .IsChecked}}invisible{{end}}">{{svg "octicon-check" 16}}</span><span class="label color" style="background-color: {{.Color}}"></span> {{.Name | RenderEmojiPlain}}
```

File: templates/repo/issue/view_content/sidebar.tmpl [hunk 7]
```diff
@@ -111,13 +111,13 @@
...
+						{{if .Description }}<br><small class="desc">{{.Description | RenderEmojiPlain}}</small>{{end}}</a>
 					{{end}}
 					<div class="ui divider"></div>
 					{{range .OrgLabels}}
-						<a class="{{if .IsChecked}}checked{{end}} item has-emoji" href="#" data-id="{{.ID}}" data-id-selector="#label_{{.ID}}"><span class="octicon-check {{if not .IsChecked}}invisible{{end}}">{{svg "octicon-check" 16}}</span><span class="label color" style="background-color: {{.Color}}"></span> {{.Name}}
-						{{if .Description }}<br><small class="desc">{{.Description}}</small>{{end}}</a>
+						<a class="{{if .IsChecked}}checked{{end}} item has-emoji" href="#" data-id="{{.ID}}" data-id-selector="#label_{{.ID}}"><span class="octicon-check {{if not .IsChecked}}invisible{{end}}">{{svg "octicon-check" 16}}</span><span class="label color" style="background-color: {{.Color}}"></span> {{.Name | RenderEmojiPlain}}
```
Comment (mrsdizzie) [inline templates/repo/issue/view_content/sidebar.tmpl]:
```suggestion
						<a class="{{if .IsChecked}}checked{{end}} item" href="#" data-id="{{.ID}}" data-id-selector="#label_{{.ID}}"><span class="octicon-check {{if not .IsChecked}}invisible{{end}}">{{svg "octicon-check" 16}}</span><span class="label color" style="background-color: {{.Color}}"></span> {{.Name | RenderEmojiPlain}}
```

File: templates/repo/issue/view_content/sidebar.tmpl [hunk 8]
```diff
@@ -128,12 +128,12 @@
 			<span class="no-select item {{if .HasSelectedLabel}}hide{{end}}">{{.i18n.Tr "repo.issues.new.no_label"}}</span>
 			{{range .Labels}}
 				<div class="item">
-					<a class="ui label has-emoji {{if not .IsChecked}}hide{{end}}" id="label_{{.ID}}" href="{{$.RepoLink}}/issues?labels={{.ID}}" style="color: {{.ForegroundColor}}; background-color: {{.Color}}" title="{{.Description}}">{{.Name}}</a>
+					<a class="ui label has-emoji {{if not .IsChecked}}hide{{end}}" id="label_{{.ID}}" href="{{$.RepoLink}}/issues?labels={{.ID}}" style="color: {{.ForegroundColor}}; background-color: {{.Color}}" title="{{.Description}}">{{.Name | RenderEmojiPlain}}</a>
```
Comment (mrsdizzie) [inline templates/repo/issue/view_content/sidebar.tmpl]:
```suggestion
					<a class="ui label {{if not .IsChecked}}hide{{end}}" id="label_{{.ID}}" href="{{$.RepoLink}}/issues?labels={{.ID}}" style="color: {{.ForegroundColor}}; background-color: {{.Color}}" title="{{.Description}}">{{.Name | RenderEmojiPlain}}</a>
```

File: templates/repo/issue/view_content/sidebar.tmpl [hunk 9]
```diff
@@ -128,12 +128,12 @@
...
 				</div>
 			{{end}}
 			{{range .OrgLabels}}
 				<div class="item">
-					<a class="ui label has-emoji {{if not .IsChecked}}hide{{end}}" id="label_{{.ID}}" href="{{$.RepoLink}}/issues?labels={{.ID}}" style="color: {{.ForegroundColor}}; background-color: {{.Color}}" title="{{.Description}}">{{.Name}}</a>
+					<a class="ui label has-emoji {{if not .IsChecked}}hide{{end}}" id="label_{{.ID}}" href="{{$.RepoLink}}/issues?labels={{.ID}}" style="color: {{.ForegroundColor}}; background-color: {{.Color}}" title="{{.Description}}">{{.Name | RenderEmojiPlain}}</a>
```
Comment (mrsdizzie) [inline templates/repo/issue/view_content/sidebar.tmpl]:
```suggestion
					<a class="ui label {{if not .IsChecked}}hide{{end}}" id="label_{{.ID}}" href="{{$.RepoLink}}/issues?labels={{.ID}}" style="color: {{.ForegroundColor}}; background-color: {{.Color}}" title="{{.Description}}">{{.Name | RenderEmojiPlain}}</a>
```

File: templates/user/dashboard/issues.tmpl [hunk 10]
```diff
@@ -123,7 +123,7 @@
 								especially on mobile views. */}}
 								<span style="line-height: 2.5">
 									{{range .}}
-										<a class="ui label has-emoji" href="{{$.Link}}?q={{$.Keyword}}&type={{$.ViewType}}&state={{$.State}}&labels={{.ID}}&milestone={{$.MilestoneID}}&assignee={{$.AssigneeID}}&repos=[{{range $.RepoIDs}}{{.}}%2C{{end}}]" style="color: {{.ForegroundColor}}; background-color: {{.Color}}" title="{{.Description}}">{{.Name}}</a>
+										<a class="ui label has-emoji" href="{{$.Link}}?q={{$.Keyword}}&type={{$.ViewType}}&state={{$.State}}&labels={{.ID}}&milestone={{$.MilestoneID}}&assignee={{$.AssigneeID}}&repos=[{{range $.RepoIDs}}{{.}}%2C{{end}}]" style="color: {{.ForegroundColor}}; background-color: {{.Color}}" title="{{.Description}}">{{.Name | RenderEmojiPlain}}</a>
```
Comment (mrsdizzie) [inline templates/user/dashboard/issues.tmpl]:
```suggestion
										<a class="ui label" href="{{$.Link}}?q={{$.Keyword}}&type={{$.ViewType}}&state={{$.State}}&labels={{.ID}}&milestone={{$.MilestoneID}}&assignee={{$.AssigneeID}}&repos=[{{range $.RepoIDs}}{{.}}%2C{{end}}]" style="color: {{.ForegroundColor}}; background-color: {{.Color}}" title="{{.Description}}">{{.Name | RenderEmojiPlain}}</a>
```

**Comment by mrsdizzie** [comment]
@jolheiser removed : )

**Review by zeripath — APPROVED** [review]

**Comment by jolheiser** [comment]
Only because it still affects labels specifically, could you also fix [`templates/repo/issue/new_form`](https://github.com/go-gitea/gitea/blob/1bf9e44bda5c8cd1fd72622cffce8ec291db79c5/templates/repo/issue/new_form.tmpl#L57-L69)?

**Comment by mrsdizzie** [comment]
oh yea missed that one (thought it was shared with view issue), fixed

**Review by jolheiser — APPROVED** [review]

**Comment by zeripath** [comment]
make lg-tm work

**Comment by guillep2k** [comment]
I marked this `skip-changelog` as I believe this regression did not affect 1.11.

**Pull request merged**